api/api_sources/sources/database/migrations/1585864510614-MechanicalTreatmentObservationSchema.ts

```diff
@@ -0,0 +1,39 @@
+import {MigrationInterface, QueryRunner} from 'typeorm';
+import { MechanicalTreatmentSchema, MechanicalTreatmentObservationSchema } from '../database-schema';
+import { AppDBMigrator } from '../applicationSchemaInterface';
+
+export class MechanicalTreatmentObservationSchema1585864510614 extends AppDBMigrator implements MigrationInterface {
+
+    mechanicalTreatmentSchema: MechanicalTreatmentSchema;
+    mechanicalTreatmentObservationSchema: MechanicalTreatmentObservationSchema;
+    /**
+     * Setup
+     */
+    setup() {
+        // Adding Water craft risk assessment init schema to migrator
+        this.mechanicalTreatmentSchema = new MechanicalTreatmentSchema();
+        this.mechanicalTreatmentObservationSchema = new MechanicalTreatmentObservationSchema();
+
+        // Modifying existing table
+        this.addSchemaVersion(this.mechanicalTreatmentSchema, 'observations');
+
+        // Creating table
+        this.addSchemaInitVersion(this.mechanicalTreatmentObservationSchema);
+    }
+
+    public async up(queryRunner: QueryRunner): Promise<any> {
+        this.log('[START]', 'UP');
+        // Running all up migration files
+        await this.runQuerySqlFiles(this.upMigrations(), queryRunner);
+        this.log('[END]', 'UP');
+    }
+
+    public async down(queryRunner: QueryRunner): Promise<any> {
+        this.log('[STAR]', 'DOWN');
+        await queryRunner.query(`DELETE FROM mechanical_treatment`);
```

Comment:
I want to explain why I added this delete query here. The `upMigrations()` removed the observation column from the table whereas the `downMigrations()` here is trying to add the `observation_id` back to the table which is a NOT NULL field. 

The migration revert will fail if there are any existing  records in the `mechanical_treatment` table. The revert will try to add the field back and the existing records have no option but to assign null for `observation_id` which is not permitted as per the schema definition. Since we are dropping all the data everytime we deploy, I don't think it is a big issue emptying the table here